crates/http/src/spin.rs

```diff
@@ -32,16 +32,32 @@ impl HttpExecutor for SpinHttpExecutor {
             component
         );
 
-        let (redirects, outputs) = capture_io_to_memory(follow, follow);
+        let mior: Option<ModuleIoRedirects> = if follow {
```

Comment:
What I'm wondering is: could we do this logic at the host level, and pass the result down to the executor from outside.  But with the current design I don't think we can - it would require significant surgery on the component and configuration objects that I don't think we should attempt in this PR.

Another possibility is to write a helper that encapsulates this pattern, but then where do we put it - if we put it in the engine crate then the encapsulation is nice but it might be too much moving policy back down to that level again.  We could put it in the trigger crate but I'm not sure that fits.

So on reflection maybe we just accept the repeated code for now and figure out how to clean it up in a subsequent PR...?

Reply:
I was thinking about this while writing this re-design to fit your spec better, and couldn't really see a way to move forward w/ the current design too...

Also, no worries about being handwavey~ I appreciate the discussion! In the end, do you want me to mark this comment as resolved and leave it for a future PR or...?

Reply:
Ah, mark it resolved.  I'll wake up at 4am one night and worry about it then. _grin_